packages/headless/src/controllers/insight/attached-results/headless-attached-results.test.ts

```diff
@@ -0,0 +1,227 @@
+import {configuration} from '../../../app/common-reducers.js';
+import {
+  attachResult,
+  detachResult,
+} from '../../../features/attached-results/attached-results-actions.js';
+import {
+  logCaseAttach,
+  logCaseDetach,
+} from '../../../features/attached-results/attached-results-analytics-actions.js';
+import {attachedResultsReducer} from '../../../features/attached-results/attached-results-slice.js';
+import {buildAttachedResultFromSearchResult} from '../../../features/attached-results/attached-results-utils.js';
+import type {InsightAppState} from '../../../state/insight-app-state.js';
+import {createMockAttachedResult} from '../../../test/mock-attached-results.js';
+import {
+  buildMockInsightEngine,
+  type MockedInsightEngine,
+} from '../../../test/mock-engine-v2.js';
+import {buildMockInsightState} from '../../../test/mock-insight-state.js';
+import {buildMockResult} from '../../../test/mock-result.js';
+import {
+  type AttachedResults,
+  type AttachedResultsOptions,
+  buildAttachedResults,
+} from './headless-attached-results.js';
+
+vi.mock(
+  '../../../features/attached-results/attached-results-analytics-actions',
+  () => ({
+    logCaseAttach: vi.fn(() => () => {}),
+    logCaseDetach: vi.fn(() => () => {}),
+  })
+);
+
+vi.mock('../../../features/attached-results/attached-results-actions');
+
+describe('insight attach to case', () => {
+  let engine: MockedInsightEngine;
+  let defaultOptions: AttachedResultsOptions;
+  let state: InsightAppState;
+  let controller: AttachedResults;
+
+  const payload = buildMockResult({
+    raw: {
+      permanentid: 'testPermanentId',
+      urihash: 'testUriHash',
+    },
+  });
+
+  function initAttachedResults(options = defaultOptions) {
+    engine = buildMockInsightEngine(buildMockInsightState());
+    engine.state = state;
+    controller = buildAttachedResults(engine, {options});
+  }
+
+  beforeEach(() => {
+    state = buildMockInsightState();
+    defaultOptions = {
+      caseId: 'defaultCaseId',
```

Comment:
same here using the value `defaultCaseId ` could give the impression that the controller uses this value by default while it's not this is an example case id,